internal/servers/repository_worker.go

```diff
@@ -292,22 +292,30 @@ func (r *Repository) UpsertWorkerStatus(ctx context.Context, worker *Worker, opt
 				case 1:
 					break
 				default:
-					return errors.New(ctx, errors.MultipleRecords, op, fmt.Sprintf("multiple records found when updating worker with id %q", worker.GetPublicId()))
+					return errors.New(ctx, errors.MultipleRecords, op, fmt.Sprintf("multiple records found when updating worker with id %q", workerClone.GetPublicId()))
 				}
 
-			case worker.GetName() != "":
-				worker.Type = KmsWorkerType.String()
+			case workerClone.GetName() != "":
+				workerClone.Type = KmsWorkerType.String()
 				workerCreateConflict := &db.OnConflict{
 					Target: db.Columns{"public_id"},
-					Action: append(db.SetColumns([]string{"description", "address"}),
+					Action: append(db.SetColumns([]string{"address"}),
 						db.SetColumnValues(map[string]interface{}{"last_status_time": "now()"})...),
 				}
 				var withRowsAffected int64
-				err := w.Create(ctx, worker, db.WithOnConflict(workerCreateConflict), db.WithReturnRowsAffected(&withRowsAffected),
+				err := w.Create(ctx, workerClone, db.WithOnConflict(workerCreateConflict), db.WithReturnRowsAffected(&withRowsAffected),
 					// The intent of this WithWhere option is to operate with the OnConflict such that the action
 					// taken by the OnConflict only applies if the conflict is on a row that is returned by this where
 					// statement, otherwise it should error out.
 					db.WithWhere("server_worker.type = 'kms'"))
+				if err == nil && workerClone.Description != worker.Description {
+					workerDescConflict := &db.OnConflict{
+						Target: db.Columns{"public_id"},
+						Action: append(db.SetColumns([]string{"description"}),
+							db.SetColumnValues(map[string]interface{}{"last_status_time": "now()"})...),
+					}
+					err = w.Create(ctx, workerClone, db.WithOnConflict(workerDescConflict))
```

Comment:
At this point you'd always know that a resource is created, so shouldn't this be an Update instead? That way you can skip the conflict stuff.

Reply:
Done